path/path.go

```diff
@@ -448,7 +448,7 @@ func (s *Service) unescapeKey(key string) string {
 
 func containsString(list []string, item string) bool {
 	for _, l := range list {
-		if l == item {
+		if strings.Index(l, item) != -1 {
```

Comment:
I am not sure this is correct. This is more like a prefix assertion. Why do we want this? If this is what we want we should rename the method because it is doing something different now.

Reply:
imho, this does actual `containssString`, previously it checked equality, which was not valid for the next case:

```
{
  "k1": [
    {
      "k2": "v2",
      "k3": {
        "k4": "v4"
       }
    }
  ]
}
// -ignoredFields=k2,k4
```

With equality check you will get `k1.[0].k2 != k2` and `k1[0].k3.k4 != k4`

Reply:
I think we should then rather provide the proper input to `containsString`. I can imagine this produces something like `k2.[0].k1 == k2`, which is not desired.

Reply:
Proper input is tricky a bit. From what I saw in examples - ignoredFields are passed and selectedFields - with whole path. I changed `Validate` to trim prefix for ignored fields